### ISSUE TITLE
First impressions... help ? :)

### ISSUE THREAD

**Issue by nvuillam** [test-zeus-ai/testzeus-hercules#11]
Hi, I spent some time installing then executing a "hello world"

1. Sorry but... Windows ? ^^
If you want to target "non technical" users (like testers ^^), the tool must be easy to install and run on Windows (Admin population do not use linux, and neither do most SF developers)
So in my opinion, the installation instructions, and demo videos, should all be on Windows (and not use VIM because you can't promote a "no-code" solution while telling to run command lines or use VIM to edit files ^^ )

After some struggle (Use Windows store to install Python3.11 then update Windows ENV vars then reboot), I succeeded to install the toolkit

2. Try to login with Salesforce

I created the hercules folders and copy-pasted the Salesforce example

![image](https://github.com/user-attachments/assets/4334cb1d-9ac3-4b39-be9f-619ff31f876b)

Then I run the command to run the test

`HEADLESS=false testzeus-hercules --project-base hercules --llm-model gpt-4o --llm-model-api-key sk-proj-xxxxxxxx`

I have a first error, but if I click ok the execution continues

![image](https://github.com/user-attachments/assets/d8b1def8-ad41-447d-9f3d-da3eebcd2022)

It opens the browser at login.salesforce.com , but then it fails

![image](https://github.com/user-attachments/assets/56c86590-8d89-444e-b730-81f98a46eea5)

Full log:

```sh
$ HEADLESS=false testzeus-hercules --project-base hercules --llm-model gpt-4o --llm-model-api-key sk-proj-xxxxx
[2024-12-05 22:15:38] INFO {state_handler.py:72} - Retrieving current state. State length: 0
[2024-12-05 22:15:38] INFO {state_handler.py:88} - Retrieving current context. State length: 0
C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)

                                                 /$$      /$$/$$                 /$$
                                                | $$  /$ | $| $$                | $$
                                                | $$ /$$$| $| $$$$$$$  /$$$$$$ /$$$$$$
                                                | $$/$$ $$ $| $$__  $$|____  $|_  $$_/
                 ==+=+####                      | $$$$_  $$$| $$  \ $$ /$$$$$$$ | $$
                +=+ ##  ##                      | $$$/ \  $$| $$  | $$/$$__  $$ | $$ /$$
                #*  %*+ ###                     | $$/   \  $| $$  | $|  $$$$$$$ |  $$$$/
            #----=+**##***#**##%%               |__/     \__|__/  |__/\_______/  \___/
            #===============------*=*
            *=====+%#%+===========*==#
            *===##%###%###*=======*=+-=
            +==##=*#%@%%%##%======+=*=--+         /$$
          ##==+%-+%@@#*%%%%%#=======*===-+       |__/
         *+*==*#+#@@%---@%%##====+==*=*****#      /$$ /$$$$$$$             /$$$$$$/$$$$ /$$   /$$
          +*==+%=*%@@*=%%%%##====+==+*===*##     | $$/$$_____/            | $$_  $$_  $| $$  | $$
           #===#####%@%##%#%=====+=+=**+*###     | $|  $$$$$$             | $$ \ $$ \ $| $$  | $$
           *====+#%%##%#%%#======+=+==#**#%%     | $$\____  $$            | $$ | $$ | $| $$  | $$
           +=========++==========+=+==#**##%     | $$/$$$$$$$/            | $$ | $$ | $|  $$$$$$$
           +=====================+=+==***##%     |__|_______/             |__/ |__/ |__/\____  $$
           %#++++++++***+===++==+==+==***#%%                                           /$$  | $$
            %**%%*+***************#+==+***%%                                          |  $$$$$$/
            %#*%#% %**********###*++++=#**%%@                                          \______/
            ###%%%#%**********######++*%**#%@
           %++=+###@**********####%   #+++*##
            #***##% #*********####    #+++#%#     /$$$$$$ /$$   /$$ /$$$$$$  /$$$$$$  /$$$$$$  /$$$$$$$ /$$$$$$
            %#**%%  #*********####    %#*#@%%    /$$__  $| $$  | $$/$$__  $$/$$__  $$/$$__  $$/$$_____//$$__  $$
            %#**%%    @%%%%%%%%%%%    %**#%%    | $$  \ $| $$  | $| $$  \__| $$  \ $| $$  \ $|  $$$$$$| $$$$$$$$
            %**#%      %####%##%      #**##%    | $$  | $| $$  | $| $$     | $$  | $| $$  | $$\____  $| $$_____/
            %**%%      %####%##%      #**##%    | $$$$$$$|  $$$$$$| $$     | $$$$$$$|  $$$$$$//$$$$$$$|  $$$$$$$
            #*#%%      #####%##%      #**%%     | $$____/ \______/|__/     | $$____/ \______/|_______/ \_______/
            #*##%       ####%##@      #**%%     | $$                       | $$
            #*#%@       ####%##@      #*#%@     | $$                       | $$
           %#*%%        %###%%%%      #*%%      |__/                       |__/
           +##%@        %###%%%       ###*#      /$$$$
           +*+*         %###%#%       @%*+#     /$$  $$
           **#+**#% @   %###%#%    %%%%%*+#     |__/\ $$
                 @@@@@@@%###%#%   #*++++#           /$$/
               @@@@@@@@@@%###%#%                   /$$/
             @@@@@@@@@@%+=####%##==++*%@@%        |__/
         @@@@@@@@@@@*=====++*%*=====%@@@@@@@@     /$$
      @@@@@@@@@@@*===============*@@@@@@@@@@@    |__/
     @@@@@@@@@*================%@@@@@@@@@+%@@
     @@@@@@@@+=++*+=========*%@@@@@@@@%%%%@@@
     @@@@@@@%**+=====++*++%@@@@@@@@@@*%@@@@
     @@@@@@@%###****++==+@@@@@@@@@@%%%@@@       /$$$$$$$                  /$$$$$$  /$$$$$$  /$$
         @@@ %%%#####**@@@@@@@@%*#@@@@         | $$__  $$                /$$__  $$/$$__  $|  $$
                   %%%#@@@@@@@%%%%@@           | $$  \ $$ /$$$$$$       | $$  \ $| $$  \ $|  $$
                       @@@@@@@@@@@             | $$  | $$/$$__  $$      | $$  | $| $$$$$$$|  $$
                        @@@@@@@                | $$  | $| $$  \ $$      | $$  | $| $$__  $| __/
                                               | $$  | $| $$  | $$      | $$/$$ $| $$  | $$
                                               | $$$$$$$|  $$$$$$/      |  $$$$$$| $$  | $$ /$$
                                               |_______/ \______/        \____ $$|__/  |__| __/
                                                                                \__/

[2024-12-05 22:15:39] INFO {__main__.py:63} - Running testcase: Successfully_create_a_new_account
[2024-12-05 22:15:39] INFO {__main__.py:64} - testcase details: Feature: Account Creation in Salesforce #newline#  #newline# Scenario:Successfully create a new accountSuccessfully create a new account #newline#  Given I am on the Salesforce login page #newline#  When I enter my username "user@example.com" and password "securePassword" #newline#  And I click on the "Log In" button #newline#  And I navigate to the "Accounts" tab #newline#  And I click on the "New" button #newline#  And I fill in the "Account Name" field with "Test Account" #newline#  And I select the "Account Type" as "Customer" #newline#  And I fill in the "Website" field with "www.testaccount.com" #newline#  And I fill in the "Phone" field with "123-456-7890" #newline#  And I click on the "Save" button #newline#  Then I should see a confirmation message "Account Test Account created successfully" #newline#  And I should see "Test Account" listed in the account records
[2024-12-05 22:15:39] INFO {agents_llm_config.py:77} - Loading configuration from environment variables
>>> Creating AutogenSimpleWrapper, Planner max chat rounds: 50, browser nav max chat rounds: 10. Save chat logs to files: True
[2024-12-05 22:15:39] INFO {config.py:239} - Created source_log_folder_path folder at: hercules\log_files\Successfully_create_a_new_account
>>> Created browser_nav_executor_agent: <testzeus_hercules.utils.autogen_sequential_function_call.UserProxyAgent_SequentialFunctionExecution object at 0x000002AA4EEFDD50>
[2024-12-05 22:15:39] INFO {base_nav_agent.py:45} - Nav agent browser_navigation_agent using model: gpt-4o
[2024-12-05 22:15:41] INFO {base_nav_agent.py:116} - Registered additional tool: store_data
[2024-12-05 22:15:41] INFO {base_nav_agent.py:116} - Registered additional tool: captcha_solver
[2024-12-05 22:15:42] INFO {base_nav_agent.py:116} - Registered additional tool: select_option
[2024-12-05 22:15:42] INFO {base_nav_agent.py:116} - Registered additional tool: bulk_select_option
[2024-12-05 22:15:42] INFO {base_nav_agent.py:116} - Registered additional tool: set_date_time_value
[2024-12-05 22:15:42] INFO {base_nav_agent.py:116} - Registered additional tool: bulk_set_date_time_value
[2024-12-05 22:15:43] INFO {base_nav_agent.py:116} - Registered additional tool: setslider
[2024-12-05 22:15:43] INFO {base_nav_agent.py:116} - Registered additional tool: bulk_set_slider
[2024-12-05 22:15:43] INFO {base_nav_agent.py:116} - Registered additional tool: upload_file
[2024-12-05 22:15:43] INFO {base_nav_agent.py:116} - Registered additional tool: bulk_upload_file
>>> Created api_nav_executor_agent: <testzeus_hercules.utils.autogen_sequential_function_call.UserProxyAgent_SequentialFunctionExecution object at 0x000002AA4EF75CD0>
[2024-12-05 22:15:43] INFO {base_nav_agent.py:45} - Nav agent api_navigation_agent using model: gpt-4o
[2024-12-05 22:15:44] INFO {base_nav_agent.py:116} - Registered additional tool: store_data
[2024-12-05 22:15:44] INFO {base_nav_agent.py:116} - Registered additional tool: create_resource_http_api
[2024-12-05 22:15:44] INFO {base_nav_agent.py:116} - Registered additional tool: read_resource_http_api
[2024-12-05 22:15:44] INFO {base_nav_agent.py:116} - Registered additional tool: update_resource_http_api
[2024-12-05 22:15:45] INFO {base_nav_agent.py:116} - Registered additional tool: patch_resource_http_api
[2024-12-05 22:15:45] INFO {base_nav_agent.py:116} - Registered additional tool: delete_resource_http_api
>>> Created api_nav_executor_agent: <testzeus_hercules.utils.autogen_sequential_function_call.UserProxyAgent_SequentialFunctionExecution object at 0x000002AA4CCBACD0>
[2024-12-05 22:15:45] INFO {base_nav_agent.py:45} - Nav agent sec_nav_agent using model: gpt-4o
[2024-12-05 22:15:45] INFO {base_nav_agent.py:116} - Registered additional tool: common_vulnerabilities_and_expos
[2024-12-05 22:15:46] INFO {base_nav_agent.py:116} - Registered additional tool: admin_panels_or_dashboards
[2024-12-05 22:15:46] INFO {base_nav_agent.py:116} - Registered additional tool: a_popular_cms_frequently_targete
[2024-12-05 22:15:46] INFO {base_nav_agent.py:116} - Registered additional tool: the_accidental_or_intentional_di
[2024-12-05 22:15:46] INFO {base_nav_agent.py:116} - Registered additional tool: cross_site_scripting
[2024-12-05 22:15:46] INFO {base_nav_agent.py:116} - Registered additional tool: open_source_intelligence
[2024-12-05 22:15:47] INFO {base_nav_agent.py:116} - Registered additional tool: technology_stacks_or_implementat
[2024-12-05 22:15:47] INFO {base_nav_agent.py:116} - Registered additional tool: misconfigurations_caused_by_impr
[2024-12-05 22:15:47] INFO {base_nav_agent.py:116} - Registered additional tool: local_file_inclusion
[2024-12-05 22:15:47] INFO {base_nav_agent.py:116} - Registered additional tool: remote_code_execution
[2024-12-05 22:15:48] INFO {base_nav_agent.py:116} - Registered additional tool: exploit_db
[2024-12-05 22:15:48] INFO {base_nav_agent.py:116} - Registered additional tool: packet_storm_security
[2024-12-05 22:15:48] INFO {base_nav_agent.py:116} - Registered additional tool: development_and_operations_pipel
[2024-12-05 22:15:48] INFO {base_nav_agent.py:116} - Registered additional tool: sql_injection
[2024-12-05 22:15:49] INFO {base_nav_agent.py:116} - Registered additional tool: cloud_infrastructure_requiring_m
[2024-12-05 22:15:49] INFO {base_nav_agent.py:116} - Registered additional tool: unauthorized_access
[2024-12-05 22:15:49] INFO {base_nav_agent.py:116} - Registered additional tool: authenticated_testing
[2024-12-05 22:15:49] INFO {base_nav_agent.py:116} - Registered additional tool: intrusive_methods_actively_explo
>>> Created sql_nav_executor_agent: <testzeus_hercules.utils.autogen_sequential_function_call.UserProxyAgent_SequentialFunctionExecution object at 0x000002AA4F05B010>
[2024-12-05 22:15:49] INFO {base_nav_agent.py:45} - Nav agent sql_navigation_agent using model: gpt-4o
[2024-12-05 22:15:50] INFO {base_nav_agent.py:116} - Registered additional tool: store_data
[2024-12-05 22:15:50] INFO {base_nav_agent.py:116} - Registered additional tool: execute_select_query_sql_async
[2024-12-05 22:15:50] INFO {high_level_planner_agent.py:46} - Planner agent using model: gpt-4o
[2024-12-05 22:15:50] INFO {config.py:271} - Created screen_shot_path folder at: hercules\proofs\Successfully_create_a_new_account
[2024-12-05 22:15:51] INFO {playwright_manager.py:301} - Launching chromium with video recording enabled.
[2024-12-05 22:15:55] INFO {runner.py:75} - Received command: Feature: Account Creation in Salesforce #newline#  #newline# Scenario:Successfully create a new accountSuccessfully create a new account #newline#  Given I am on the Salesforce login page #newline#  When I enter my username "user@example.com" and password "securePassword" #newline#  And I click on the "Log In" button #newline#  And I navigate to the "Accounts" tab #newline#  And I click on the "New" button #newline#  And I fill in the "Account Name" field with "Test Account" #newline#  And I select the "Account Type" as "Customer" #newline#  And I fill in the "Website" field with "www.testaccount.com" #newline#  And I fill in the "Phone" field with "123-456-7890" #newline#  And I click on the "Save" button #newline#  Then I should see a confirmation message "Account Test Account created successfully" #newline#  And I should see "Test Account" listed in the account records
[2024-12-05 22:15:55] INFO {runner.py:86} - Processing command: Feature: Account Creation in Salesforce #newline#  #newline# Scenario:Successfully create a new accountSuccessfully create a new account #newline#  Given I am on the Salesforce login page #newline#  When I enter my username "user@example.com" and password "securePassword" #newline#  And I click on the "Log In" button #newline#  And I navigate to the "Accounts" tab #newline#  And I click on the "New" button #newline#  And I fill in the "Account Name" field with "Test Account" #newline#  And I select the "Account Type" as "Customer" #newline#  And I fill in the "Website" field with "www.testaccount.com" #newline#  And I fill in the "Phone" field with "123-456-7890" #newline#  And I click on the "Save" button #newline#  Then I should see a confirmation message "Account Test Account created successfully" #newline#  And I should see "Test Account" listed in the account records
[2024-12-05 22:15:55] INFO {autogen_simple_wrapper.py:751} - Prompt for command: Execute the user task "Feature: Account Creation in Salesforce #newline#  #newline# Scenario:Successfully create a new accountSuccessfully create a new account #newline#  Given I am on the Salesforce login page #newline#  When I enter my username "user@example.com" and password "securePassword" #newline#  And I click on the "Log In" button #newline#  And I navigate to the "Accounts" tab #newline#  And I click on the "New" button #newline#  And I fill in the "Account Name" field with "Test Account" #newline#  And I select the "Account Type" as "Customer" #newline#  And I fill in the "Website" field with "www.testaccount.com" #newline#  And I fill in the "Phone" field with "123-456-7890" #newline#  And I click on the "Save" button #newline#  Then I should see a confirmation message "Account Test Account created successfully" #newline#  And I should see "Test Account" listed in the account records"
user (to planner_agent):

Execute the user task "Feature: Account Creation in Salesforce #newline#  #newline# Scenario:Successfully create a new accountSuccessfully create a new account #newline#  Given I am on the Salesforce login page #newline#  When I enter my username "user@example.com" and password "securePassword" #newline#  And I click on the "Log In" button #newline#  And I navigate to the "Accounts" tab #newline#  And I click on the "New" button #newline#  And I fill in the "Account Name" field with "Test Account" #newline#  And I select the "Account Type" as "Customer" #newline#  And I fill in the "Website" field with "www.testaccount.com" #newline#  And I fill in the "Phone" field with "123-456-7890" #newline#  And I click on the "Save" button #newline#  Then I should see a confirmation message "Account Test Account created successfully" #newline#  And I should see "Test Account" listed in the account records"

--------------------------------------------------------------------------------
planner_agent (to user):

{
    "plan": "1. Navigate to the Salesforce login page.\n2. Enter the username 'user@example.com' and password 'securePassword'.\n3. Click on the 'Log In' button.\n4. Navigate to the 'Accounts' tab.\n5. Click on the 'New' button to create a new account.\n6. Fill in the 'Account Name' field with 'Test Account'.\n7. Select the 'Account Type' as 'Customer'.\n8. Fill in the 'Website' field with 'www.testaccount.com'.\n9. Fill in the 'Phone' field with '123-456-7890'.\n10. Click on the 'Save' button.\n11. Verify that a confirmation message 'Account Test Account created successfully' is displayed.\n12. Verify that 'Test Account' is listed in the account records.",
    "next_step": "Navigate to the Salesforce login page.",
    "terminate": "no",
    "is_assert": false,
    "is_passed": true,
    "target_helper": "browser"
}

--------------------------------------------------------------------------------
No listeners available, discarding message: {'message': "Plan:\n1. Navigate to the Salesforce login page.\n 2. Enter the username 'user@example.com' and password 'securePassword'.\n 3. Click on the 'Log In' button.\n 4. Navigate to the 'Accounts' tab.\n 5. Click on the 'New' button to create a new account.\n 6. Fill in the 'Account Name' field with 'Test Account'.\n 7. Select the 'Account Type' as 'Customer'.\n 8. Fill in the 'Website' field with 'www.testaccount.com'.\n 9. Fill in the 'Phone' field with '123-456-7890'.\n 10. Click on the 'Save' button.\n 11. Verify that a confirmation message 'Account Test Account created successfully' is displayed.\n 12. Verify that 'Test Account' is listed in the account records.", 'type': 'plan'}       
No listeners available, discarding message: {'message': 'Next step: Navigate to the Salesforce login page.', 'type': 'step'}

********************************************************************************
Starting a new chat....

********************************************************************************
user (to chat_manager):

Navigate to the Salesforce login page. Current Page: about:blank, Title:  ##target_helper: browser##

--------------------------------------------------------------------------------

Next speaker: browser_navigation_agent


>>>>>>>> USING AUTO REPLY...
browser_navigation_agent (to chat_manager):

***** Suggested tool call (call_AUZ3mSgDimKthhE0aEO3gu9M): openurl *****
Arguments:
{"url":"https://login.salesforce.com"}
************************************************************************

--------------------------------------------------------------------------------

Next speaker: browser_nav_executor


>>>>>>>> EXECUTING ASYNC FUNCTION openurl...
[2024-12-05 22:16:05] INFO {open_url.py:28} - Opening URL: https://login.salesforce.com
No listeners available, discarding message: {'message': 'Opened URL: https://login.salesforce.com', 'type': 'action'}
browser_nav_executor (to chat_manager):

***** Response from calling tool (call_AUZ3mSgDimKthhE0aEO3gu9M) *****
Page loaded: https://login.salesforce.com/, Title: Connexion | Salesforce
**********************************************************************

--------------------------------------------------------------------------------

Next speaker: browser_navigation_agent


>>>>>>>> USING AUTO REPLY...
browser_navigation_agent (to chat_manager):

##TERMINATE TASK##

--------------------------------------------------------------------------------
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\Scripts\testzeus-hercules.exe\__main__.py", line 7, in <module>
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testzeus_hercules\__main__.py", line 179, in main
    sequential_process()
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testzeus_hercules\__main__.py", line 69, in sequential_process
    asyncio.run(runner.start())
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\nest_asyncio.py", line 30, in run
    return loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\nest_asyncio.py", line 98, in run_until_complete
    return f.result()
           ^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\asyncio\futures.py", line 203, in result 
    raise self._exception.with_traceback(self._exception_tb)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\asyncio\tasks.py", line 277, in __step   
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testzeus_hercules\core\runner.py", line 206, in start
    self.result, self.execution_time = await self.process_command(self.command)
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testzeus_hercules\core\runner.py", line 89, in process_command
    result = await self.autogen_wrapper.process_command(command, current_url)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testzeus_hercules\core\autogen_simple_wrapper.py", line 757, in process_command
    result = await self.agents_map["user"].a_initiate_chat(  # type: ignore
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\autogen\agentchat\conversable_agent.py", line 1162, in a_initiate_chat
    msg2send = await self.a_generate_reply(messages=self.chat_messages[recipient], sender=recipient)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\autogen\agentchat\conversable_agent.py", line 2134, in a_generate_reply
    final, reply = reply_func(self, messages=messages, sender=sender, config=reply_func_tuple["config"])
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\autogen\agentchat\conversable_agent.py", line 508, in wrapped_reply_func
    return reply_func_from_nested_chats(chat_queue, recipient, messages, sender, config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\autogen\agentchat\conversable_agent.py", line 433, in _summary_from_nested_chats
    res = initiate_chats(chat_to_run)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\autogen\agentchat\chat.py", line 217, in initiate_chats
    chat_res = sender.initiate_chat(**chat_info)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\autogen\agentchat\conversable_agent.py", line 1107, in initiate_chat
    summary = self._summarize_chat(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\autogen\agentchat\conversable_agent.py", line 1228, in _summarize_chat
    summary = summary_method(self, recipient, summary_args)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testzeus_hercules\core\autogen_simple_wrapper.py", line 176, in my_custom_summary_method
    self.__save_chat_log(sender, recipient)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testzeus_hercules\core\autogen_simple_wrapper.py", line 373, in __save_chat_log
    with open(chat_logs_file, "w") as file:
         ^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 22] Invalid argument: 'hercules\\log_files\\Successfully_create_a_new_account\\log_between_sender-user->rec-chat_manager_2024-12-05T22:16:07.824902.json'
Sentry is attempting to send 3 pending events
Waiting up to 2 seconds
Press Ctrl-Break to quit

33614@LAPTOP-9FSG2200 MINGW64 /c/git/pro/hardis-sfdx-official (features/dev/my-task)
$ node:events:497
      throw er; // Unhandled 'error' event
      ^

Error: EPIPE: broken pipe, write
    at Socket._write (node:internal/net:61:18)
    at writeOrBuffer (node:internal/streams/writable:570:12)
    at _write (node:internal/streams/writable:499:10)
    at Writable.write (node:internal/streams/writable:508:10)
    at PipeTransport.send (C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\driver\package\lib\protocol\transport.js:51:21)
    at dispatcherConnection.onmessage (C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\driver\package\lib\cli\driver.js:59:57)
    at DispatcherConnection.sendEvent (C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\driver\package\lib\server\dispatchers\dispatcher.js:178:10)
    at BrowserContextDispatcher._dispatchEvent (C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\driver\package\lib\server\dispatchers\dispatcher.js:108:22)
    at CRBrowserContext.<anonymous> (C:\Users\33614\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\driver\package\lib\server\dispatchers\browserContextDispatcher.js:166:12)
    at CRBrowserContext.emit (node:events:519:28)
Emitted 'error' event on Socket instance at:
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
    at runNextTicks (node:internal/process/task_queues:64:3)
    at process.processImmediate (node:internal/timers:454:9) {
  errno: -4047,
  syscall: 'write',
  code: 'EPIPE'
}

Node.js v20.16.0
^C
```

Additional questions:

- Would it be possible to send parameters using CLI arguments and not ENV variables ? ( like --headless or --no-headless for example )
- How can we run a single test ?
- Would it be possible to have an output folder by test run, instead of having output files generated in multiple folders ? (logfile, output, proofs)

Conclusion:

I am really interested in your project, because if it really works it can be a revolution :) But for now, if a CTO can't make it work, "average user" won't succeed either... let's work together to improve that :)

**Comment by shriyanshagnihotri** [comment]
@nvuillam Thanks a lot for trying out this and reporting the issue.
Let me try this in a windows machine to reproduce this.

You can join our slack community to get quicker help: https://join.slack.com/t/testzeuscommunityhq/shared_invite/zt-2v2br8wog-FAmo_76xRHx~k~1oNaGQ0Q

Meanwhile some background, we are still exploring what is the best interface for the engineers to interact with testzeus-hercules. The windows path is for sure definitely more complicated over the MacOS and Linux. 

I will look into to this issue ASAP as you are right this blocks our adoption for sure.

**Comment by nvuillam** [comment]
@shriyanshagnihotri thanks for your answer :)

If you make it work on Windows from VsCode with git bash for Windows, I'll bring you many Salesforce users through a new [VsCode SFDX Hardis](https://marketplace.visualstudio.com/items?itemName=NicolasVuillamy.vscode-sfdx-hardis) menu, it's less optimal than a real UI, but it prevents to need to write command lines (they are written in the background) ^^

**Comment by shriyanshagnihotri** [comment]
@nvuillam I have tried to solve this by adding an helper PowerShell script to install the relevant packages and permissions before starting with testers-Hercules in windows.

check this: https://github.com/test-zeus-ai/testzeus-hercules?tab=readme-ov-file#%EF%B8%8F-running-hercules-on-a-windows-machine

Also we found out 3 bugs,
1) ad-blocker plugins not loading (as its not required, for now I have disabled it in windows)
2) file name of inner logs of agent was causing failure in windows as even shown in your provided logs.
3) If ffmpeg not installed in Windows, the session recording will not happen.

All of the above is resolved with the release 0.0.12 (latest)

thanks for bringing this up.

We are still exploring more interactive way to solve the UX of hercules.
